test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -53,31 +55,91 @@ struct test_inclusive_scan_by_segment
         }
     }
 
-    template <typename Iterator1, typename Iterator2, typename Size>
+    struct last_accum_state
+    {
+        int idx_begin = 0;
+        int idx_end = 0;
+        int summ = 0;
```

Comment:
1) why "int"? `inclusive_scan_by_segment` accepts  generic types.
2) why just "+"? `inclusive_scan_by_segment` accepts a BinaryOperator.

Reply:
(1) - fixed

Reply:
(2) - because only "+" used  in previous implementation of check_values

Reply:
I would ask for your attention regarding "`inclusive_scan_by_segment` accepts a BinaryOperator." That case must be covered as well. Yes, I know the test didn't have such coverage before. But,
1) As far as we have started modifying the test, lets doing it good.
2) IMPHO, we cannot it postpone, because we will have to change the same code twice.

Reply:
Fixed